android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -616,6 +621,48 @@ public void disableNativeFramesTracking() {
         }
     }
 
+    public WritableMap startProfiling() {
+        final WritableMap result = new WritableNativeMap();
+        try {
+            HermesSamplingProfiler.enable();
+            result.putBoolean("started", true);
+        } catch (Throwable e) {
+            result.putBoolean("started", false);
+            result.putString("error", e.toString());
+        }
+        return result;
+    }
+
+    public WritableMap stopProfiling() {
+        final WritableMap result = new WritableNativeMap();
+        try {
+            HermesSamplingProfiler.disable();
+
+            final File output = File.createTempFile(
+                "sampling-profiler-trace", ".cpuprofile", reactApplicationContext.getCacheDir());
+
+            if (this.debugEnabled) {
+                logger.log(SentryLevel.INFO, "Profile saved to: " + output.getAbsolutePath());
+            }
+
+            try (final BufferedReader br = new BufferedReader(new FileReader(output));) {
+                HermesSamplingProfiler.dumpSampledTraceToFile(output.getPath());
+
+                final StringBuilder text = new StringBuilder();
+                String line;
+                while ((line = br.readLine()) != null) {
+                    text.append(line);
+                    text.append('\n');
+                }
+
+                result.putString("profile", text.toString());
+            }
+        } catch (Throwable e) {
+            result.putString("error", e.toString());
```

Comment:
Should we just log in to the native plugins instead of returning the stringified error avoiding the back-and-forth of messages in the native bridges?

Reply:
From developer experience, at the moment people will see the error right away in the Metro DevServer logs, but yes, we keep the native bridge a bit more busy.